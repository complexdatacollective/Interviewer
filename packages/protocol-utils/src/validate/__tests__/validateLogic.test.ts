/* eslint-env jest */

const validateLogic = require('../validateLogic');
const invalidProtocol = require('./invalidProtocol.json');

describe('validateLogic', () => {
  it('A well formed protocol will return an array of errors', () => {
    const logicErrors = validateLogic(invalidProtocol);
    expect(logicErrors).toMatchSnapshot();
  });

  it('It can handle undefined values in the protocol', () => {
    invalidProtocol.stages[0].skipLogic.filter.rules = undefined;

    expect(() => {
      validateLogic(invalidProtocol)
    }).not.toThrow();
  });
})
