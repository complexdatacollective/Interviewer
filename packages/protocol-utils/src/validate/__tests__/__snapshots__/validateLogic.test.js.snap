// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateLogic A well formed protocol will return an array of errors 1`] = `
Array [
  "protocol.codebook: Duplicate entity name \\"du\\"",
  "protocol.stages[0].form.fields[0]: Form field variable not found in codebook.",
  "protocol.stages[5].form.fields[0]: Form field variable not found in codebook.",
]
`;
